### PR TITLE
fix: Enum Type and Enum Field Deserialization

### LINKED ISSUE #14: EnumType not deserialized correctly and fields need also check
# Background

The EnumType feature firstly is not deserialized correctly. Right there is always an empty collection.

Additionally, there should be a check to see whether the value of ab EnumType field is a valid value.

# Action

- Fix deserialization of enum type
- Add check on values of enum type field

### PR DESCRIPTION
**Pull request by zeljkobekcic**
Previously the deserialization of an enum type was not correct and also
there was no check if the values in an enum field were actually valid.
This branch fixes these two problems.

### CODE REVIEW

File: serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/deserializer/SchemaDeserializer.kt [hunk 1]
```diff
@@ -50,7 +50,7 @@ class SchemaDeserializer: StdDeserializer<Schema>(Schema::class.java) {
 
     private fun toEnumType(node: JsonNode): Type {
         val name = node.get("name").asText()
-        val values = node.findValuesAsText("values").toSet()
+        val values = node.get("values").mapValues { it.asText() }
```
Comment (Lacritz) [inline serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/deserializer/SchemaDeserializer.kt:53]: is the change necessary?

File: serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/exception/InvalidEnumValueException.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package io.datalbry.precise.serialization.jackson.exception
+
+import io.datalbry.precise.api.schema.type.EnumType
+
+class InvalidEnumValueException(type: EnumType, value: String) :
+    IllegalArgumentException("Value \"${value}\" is not contained by type \"${type.name}\" with values: ${type.values}")
```
Comment (Lacritz) [inline serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/exception/InvalidEnumValueException.kt]: We are logging values using the following format `[<message>]`

Comment (Lacritz) [reply serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/exception/InvalidEnumValueException.kt]: `Value \"${value}\" is not contained by type \"${type.name}\" with values: ${type.values}`

should be the following:

`Value[${value}] is not contained by EnumType[${type.name}] with values ${type.values}`

Comment (Lacritz) [reply serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/exception/InvalidEnumValueException.kt]: + missing new line

File: serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/deserializer/SchemaDeserializer.kt (hunk 1 above)
Comment (zeljkobekcic) [reply serialization-jackson/src/main/kotlin/io/datalbry/precise/serialization/jackson/deserializer/SchemaDeserializer.kt:53]: In this case `node.findValuesAsText("values")` returns a list with an empty string. We expect there a collection of strings that match the enum definition.

The change here `node.get("values")` returns a list of the expected values wrapped in a JsonNode.

**Pull request merged**